logp/logptest/logger.go

```diff
@@ -0,0 +1,187 @@
+//     when searching for logs
+type Logger struct {
+	*logp.Logger
+	t       *testing.T
+	logFile *os.File
+	offset  int64
+}
+
+// NewFileLogger returns a logger that logs to a file and has methods
+// to search in the logs.
+// The *logp.Logger is embedded into it, so [Logger] is a drop-in
+// replacement for a *logp.Logger, or the logger can be accessed via
+// [Logger.Logger]
+func NewFileLogger(t *testing.T) *Logger {
+	encoderConfig := ecszap.ECSCompatibleEncoderConfig(zapcore.EncoderConfig{})
+	encoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
+	encoder := zapcore.NewJSONEncoder(encoderConfig)
+
+	f, err := os.CreateTemp("", "testing-logger-*.log")
+	if err != nil {
+		t.Fatalf("cannot create log file: %s", err)
+	}
+
+	core := zapcore.NewCore(encoder, zapcore.AddSync(f), zap.DebugLevel)
+
+	tl := &Logger{
+		t:       t,
+		logFile: f,
+	}
+
+	t.Cleanup(func() {
+		// Sync the core, the file, then close the file
+		if err := core.Sync(); err != nil {
+			t.Logf("cannot sync zap core: %s", err)
+		}
+
+		if err := f.Sync(); err != nil {
+			t.Logf("cannot sync log file: %s", err)
+		}
+
+		if err := f.Close(); err != nil {
+			t.Logf("cannot close log file: %s", err)
+		}
+
+		// If the test failed, print the log file location,
+		// otherwise remove it.
+		if t.Failed() {
+			t.Logf("Full logs written to %s", f.Name())
+			return
+		}
+
+		if err := os.Remove(f.Name()); err != nil {
+			t.Logf("could not remove temporary log file: %s", err)
+		}
+	})
+
+	logger := logp.NewLogger(
+		"",
+		zap.WrapCore(func(in zapcore.Core) zapcore.Core {
+			return core
+		}))
+
+	tl.Logger = logger
+
+	return tl
+}
+
+// WaitLogsContains waits for the specified string s to be present in the logs within
+// the given timeout duration and fails the test if s is not found.
+// It keeps track of the log file offset, reading only new lines. Each
+// subsequent call to WaitLogsContains will only check logs not yet evaluated.
+// msgAndArgs should be a format string and arguments that will be printed
+// if the logs are not found, providing additional context for debugging.
+func (l *Logger) WaitLogsContains(s string, timeout time.Duration, msgAndArgs ...any) {
+	l.t.Helper()
+	require.EventuallyWithT(
+		l.t,
+		func(c *assert.CollectT) {
+			found, err := l.logContains(s)
+			if err != nil {
+				c.Errorf("cannot check the log file: %s", err)
+				return
+			}
+
+			if !found {
+				c.Errorf("did not find '%s' in the logs", s)
+			}
+		},
+		timeout,
+		100*time.Millisecond,
+		msgAndArgs...)
+}
+
+// logContains searches for str in the log file keeping track of the offset.
+// If there is any issue reading the log file, then t.Fatalf is called,
+// if str is not present in the logs, t.Fatalf is called.
+func (l *Logger) LogContains(str string) {
+	l.t.Helper()
+	found, err := l.logContains(str)
+	if err != nil {
+		l.t.Fatalf("cannot read log file: %s", err)
+	}
+
+	if !found {
+		l.t.Fatalf("'%s' not found in logs", str)
+	}
```

Comment:
```suggestion
	return assert.Truef(l.t, found, "did not find '%s' in the log", str)
```

this is not really a fatal error, users might desire to continue with the test

Reply:
If you aren't failing the test immediately, it should not be called `Require*`, but `Assert*` so it's consistent with we're used to with testify